### PR TITLE
Add GHA build and deploy workflows

### PR DESCRIPTION
**Pull request by rcooke-warwick**
Adding BalenaOS GitHub action to test on balena-intel repository

### CODE REVIEW

**Comment by rcooke-warwick** [comment]
for some reason all 3 suites fail to pull the worker container

```
worker Pulling 
worker Error 
Error response from daemon: Get "https://registry-proxy.balena-cloud.com/v2/balena/leviathan-worker-amd64/2.9.37/manifests/latest": Get "https://api.balena-cloud.com/auth/v1/token?scope=repository%3Abalena%2Fleviathan-worker-amd64%2F2.9.37%3Apull&service=registry2.balena-cloud.com": net/http: request canceled (Client.Timeout exceeded while awaiting headers)
make[1]: *** [Makefile:81: test] Error 18
make[1]: Leaving directory '/home/runner/_work/balena-intel/balena-intel/layers/meta-balena/tests/leviathan'
make: *** [Makefile:84: local-test] Error 2
```

File: .github/workflows/genericx86-64.yml [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+    uses: balena-os/balena-yocto-scripts/.github/workflows/yocto-build-deploy.yml@v1.25.3
+    # Prevent duplicate workflow executions for pull_request (PR) and pull_request_target (PRT) events.
+    # Both PR and PRT will be triggered for the same pull request, whether it is internal or from a fork.
+    # This condition will prevent the workflow from running twice for the same pull request while
+    # still allowing it to run for all other event types.
+    if: (github.event.pull_request.head.repo.full_name == github.repository) == (github.event_name == 'pull_request')
+    secrets: inherit
+    with:
+      machine: genericx86-64-ext
+      # Needed for testing - deafults to production
```
Comment (vipulgupta2048) [inline .github/workflows/genericx86-64.yml:37]:
```suggestion
      # Needed for testing - default to production
```

**Comment by klutchell** [comment]
@resin-jenkins retest this please

**Comment by klutchell** [comment]
I self-certify!

**Pull request merged**